src/MalformedDateString.php

```diff
@@ -0,0 +1,10 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\Clock;
+
+use Exception;
+
+class MalformedDateString extends Exception
```

Comment:
I'm not sure about adding an exception only for a testing thing (that will only be relevant until PHP 8.2 is kicked out)...

I'll simplify things a bit here.